db/migrate/20210226152201_create_content_groups.rb

```diff
@@ -0,0 +1,11 @@
+class CreateContentGroups < ActiveRecord::Migration[6.0]
+  def change
+    create_table :content_groups do |t|
```

Comment:
I'm a bit confused about the structure of this model? It has both a header and links. Does that mean the plan is to replace the SubSection model with a ContentGroup? 

Does the ContentGroup contain all of the content in a subsection? If so, how do we know which order the header and links should be presented in?

What happens if there is more than one header in the subsection?

Reply:
So a content group is a header and its associated links. Pretty much matches what a content group is [here](https://github.com/alphagov/collections-publisher/blob/2c3acd32c13adfa6be393adadd8d3bc5ee8a89cd/app/presenters/coronavirus/sub_section_json_presenter.rb#L55). This means that we can remove functionality from the json presenter pretty directly. 

This means that a `SubSection` has many `ContentGroups` and each content group is aware of its position relative to one another. There's also no need for a `ContentGroup` to have a heading, the only required attribute is links. 

Sorry, should have made this clear in the commit messages.

Reply:
Ok, that sort of makes sense. So would a featured link be classed as a ContentGroup that doesn't have a header?
Also, `links` are being stored as a string at the moment. Do you have plans to store them as an array?

Reply:
yeah I think we'd like to store links as an array going forward. As for the featured links we haven't given that a huge amount of thought yet. I think we need to look into it a bit more so that we can understand the best way to store information on which link is featured. As is stands a featured link is just an array item within links but we don't have any way of knowing which content group a featured link for a sub section is in, only that one exists.

Reply:
My main concern is that it's not obvious what a "Content group" is and I think that will trip people up in the future.

Also, isn't a featured link it's own content group?

Reply:
I think you might be right there, do you think a 'lines' model would be better?

Not necessarily, you could have a content group like;
```
header: "header"
links: ["[link](/link)", "[link2](/link2)"]
```
Where `/link2` is the featured link. The content group as it is has no knowledge on whether it has a featured link in the "links" section.

Reply:
If we define content_groups as what's in the content_item, I think featured links appear in their own group. As it in the [Work and financial support accordion](https://www.gov.uk/api/content/coronavirus):

```
{
   "title":"Work and financial support",
   "sub_sections":[
      {
         "list":[
            {
               "url":"/coronavirus/worker-support?priority-taxon=774cee22-d896-44c1-a611-e3109cce8eae",
               "label":"Getting financial help and staying safe at work",
               "description":"Get coronavirus (COVID-19) support if you’re working less or not working - and find out how to stay safe at work.",
               "featured_link":true
            }
         ],
         "title":null
      },
      {
         "list":[
            {
               "url":"/guidance/check-if-you-could-be-covered-by-the-coronavirus-job-retention-scheme?priority-taxon=774cee22-d896-44c1-a611-e3109cce8eae",
               "label":"Furlough: check if your employer can put you on temporary paid leave"
            },
            {
               "url":"/redundancy-your-rights?priority-taxon=774cee22-d896-44c1-a611-e3109cce8eae",
               "label":"Redundancy: your rights"
            },
            {
               "url":"/test-and-trace-support-payment?priority-taxon=774cee22-d896-44c1-a611-e3109cce8eae",
               "label":"Apply for a Test and Trace Support Payment"
            },
            {
               "url":"https://www.citizensadvice.org.uk/work/coronavirus-if-youre-worried-about-working?priority-taxon=774cee22-d896-44c1-a611-e3109cce8eae",
               "label":"Citizens Advice: advice if you’re worried about working"
            }
         ],
         "title":"Popular content"
      }
   ]
},
```

Reply:
Then again, that might be because at the moment featured links only appear at the top of the accordion.

I don't know if a `lines` models is _better_. Just different. featured links do throw a bit of a spanner in the works, but I don't think they can be ignored. I think they need to be front and centre of any modelling decision as they can be complicated.